packages/codebytes/src/index.tsx

```diff
@@ -37,31 +37,42 @@ const EditorContainer = styled(Background)<EditorStyleProps>(
 export const CodeByteEditor: React.FC<CodeByteEditorProps> = ({
   text: initialText,
   language: initialLanguage,
-  hideCopyButton,
-  isIFrame = false,
+  hideCopyButton = false,
+  isForums = false,
   snippetsBaseUrl = process.env.CONTAINER_API_BASE,
   onEdit,
   onLanguageChange,
+  onCopy,
 }) => {
-  const [text, setText] = useState<string>(initialText);
-  const [language, setLanguage] = useState<LanguageOption>(initialLanguage);
+  const [text, setText] = useState<string>(initialText ?? '');
+  const [language, setLanguage] = useState<LanguageOption>(
+    initialLanguage ?? ''
+  );
   const [hasBeenEdited, setHasBeenEdited] = useState(false);
 
   useEffect(() => {
-    const options = getOptions();
-    const page_name = options.renderMode
-      ? `${options.clientName}_${options.renderMode}`
-      : options.clientName;
+    if (isForums) {
+      const options = getOptions();
+      const page_name = options.renderMode
+        ? `${options.clientName}_${options.renderMode}`
+        : options.clientName;
+
+      trackUserImpression({
+        page_name,
+        context: options.parentPage,
+        target: 'codebyte',
+      });
+    }
+  }, [isForums]);
 
-    trackUserImpression({
-      page_name,
-      context: options.parentPage,
-      target: 'codebyte',
-    });
-  }, []);
+  useEffect(() => {
+    if (language) {
+      setText(initialText ?? (helloWorld[language] || ''));
```

Comment:
why are we adding this hook?

Reply:
From the unit tests in static-sites this intended behavior was made more clear: if the initial text is not provided, then the content is initialized with predefined text for the chosen language. The empty string at the end is for when there is not initial content for that language. 

We need it in a useEffect, because if the text and language props change, the change needs to be reapplied.

Reply:
is there a way for language to change? i see a way for someone to do that, in which case there would just be the initial state

Reply:
@borisonr  Hehe, good callout. Since the parent controls the language, and passes it in as props, the language can still _technically_ change, though it's less likely to and more of a design question. This would make it a bit more flexible, for situations where we may want to pass in a different language from the parent.

I can just add it as initial state, though, if we like!

Reply:
just trying to be mindful here that this pr was to add tests... probably shouldnt expand scope to potential future features?

Reply:
though this also ensures the desired testing behavior passes

Reply:
why are we testing language changing if we dont allow language to be changed yet?

Reply:
not necessarily a future feature, a use case that works with query params and now am ensuring works in our version

Reply:
not a blocker, just a little confused. is this what is was in static sites too?

Reply:
the test/desired behavior is in static sites, and the behavior would change as such if the query params passed in changed

Reply:
but query params would cause a page reload and still work with initial state, no?

Reply:
if we do want to do with something like initial state, we could do: 
```
const getInitialText = () => {
    if (initialText !== undefined) return initialText;
    return initialLanguage ? helloWorld[initialLanguage] : '';
};
```
and then:

```
const [text, setText] = useState<string>(getInitialText());
```
this would also pass the original test

Reply:
ok with either, using the useEffect for extensibility with props being passed, but could also go with initial state if we want to keep it as state

Reply:
got it, this works! sorry for the back and forth was just trying to understand the non-test related changes in the pr!

Reply:
all good! I'll update this pr and clear with this logic and clear out some merge conflicts!